### PR TITLE
Feat watch history

### PR DESCRIPTION
**Pull request by nakulsharma15**
Feat: Added History Page
1. Videos watched by the user will appear in the history page.
2. User can delete videos from the history page or clear all videos at once by clicking on "Clear All" button,

### CODE REVIEW

**Review by vaibhav-mahalle — APPROVED** [review]
I liked the following things in your pull request:

1. the code is committed to GitHub very systematically, according to the commit messages
2. commit messages are clear and precise

File: src/Components/HorizontalVideoCard.jsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+              headers: {
+                authorization: localStorage.getItem("Token"),
+              },
+            });
+            const { status, data } = response;
+            if (status === 200 || status === 201) {
+              userDispatch({ type: "REMOVE_FROM_HISTORY", payload: data?.history });
+            }
+          } catch (error) {
+            console.log(error);
```
Comment (rahulcs754) [inline src/Components/HorizontalVideoCard.jsx:24]: i think you can remove console log

File: src/Components/LoginForm.jsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import "./Styles/LoginForm.css";
-import { Link, useNavigate } from "react-router-dom";
+import { Link, useNavigate, useLocation } from "react-router-dom";
 import { useFormik } from "formik";
 import * as Yup from "yup";
```
Comment (rahulcs754) [inline src/Components/LoginForm.jsx:4]: nice to use formik for form validation.

File: src/Components/LoginForm.jsx [hunk 3]
```diff
@@ -26,11 +27,11 @@ export default function LoginForm() {
             setUserDetails(response.data.foundUser);
             localStorage.setItem("Token", response.data.encodedToken);
             toast.success("You're successfully logged in.", {style: toastStyle});
-            navigate(location.state?.from?.pathname || "/");
+            navigate(location.state?.from?.pathname);
           }
         } catch (error) {
           console.log(error);
```
Comment (rahulcs754) [inline src/Components/LoginForm.jsx:33]: remove console.log

**Review by rahulcs754 — APPROVED** [review]
The code looks clean and good .

just small suggestion provide

File: src/Components/HorizontalVideoCard.jsx (hunk 1 above)
Comment (nakulsharma15) [reply src/Components/HorizontalVideoCard.jsx:24]: Thanks for pointing out. I will remove it.

File: src/Components/LoginForm.jsx (hunk 2 above)
Comment (nakulsharma15) [reply src/Components/LoginForm.jsx:4]: Thank you

**Pull request merged**